src/local_scheduler/test/local_scheduler_tests.cc

```diff
@@ -234,8 +234,15 @@ TEST object_reconstruction_test(void) {
     Task *task = Task_alloc(
         execution_spec, TASK_STATUS_DONE,
         get_db_client_id(local_scheduler->local_scheduler_state->db));
+#if !RAY_USE_NEW_GCS
     task_table_add_task(local_scheduler->local_scheduler_state->db, task, NULL,
                         NULL, NULL);
+#else
+    RAY_CHECK_OK(TaskTableAdd(
+        &local_scheduler->local_scheduler_state->gcs_client, task));
+    Task_free(task);
```

Comment:
Don't these Task_free need to be placed outside of the #if block?

Reply:
task_table_add takes ownership. I know, it's surprising (but documented).